pandas/io/formats/format.py

```diff
@@ -561,7 +561,19 @@ def __init__(
         self.sparsify = sparsify
 
         self.float_format = float_format
-        self.formatters = formatters if formatters is not None else {}
+        if formatters is not None and (
+            len(frame.columns) == len(formatters) or isinstance(formatters, dict)
+        ):
+            self.formatters = formatters
+        elif formatters is not None:
+            raise ValueError(
+                (
+                    "Formatters length({flen}) should match"
+                    + " DataFrame number of columns({dlen})"
```

Comment:
the `+` is not necessary.

Reply:
I just used because there is a warning of line too long with the flake8 command. What should I do to avoid that?